bdb/bdb_net.c

```diff
@@ -220,15 +220,21 @@ char *print_addr(struct sockaddr_in *addr, char *buf)
     int rc = getnameinfo((struct sockaddr *)addr, len, name, sizeof(name),
                          service, sizeof(service), 0);
     if (rc) {
-        strerror_r(rc, errbuf, sizeof(errbuf));
+        char *tmp = strerror_r(rc, errbuf, sizeof(errbuf));
```

Comment:
These are incorrect. First of all `char *strerror_r(int errnum, char *buf, size_t buflen);` is a GNU extension. Second of all, the GNU extension returns an error string, which means `tmp` will always be non-NULL.